src/modules/site-v2/templates/primary/map.html

```diff
@@ -0,0 +1,240 @@
+    popupAnchor: [0, -15],
+});
+const icon_tropicalis = L.icon({
+    iconUrl: "{{ url_for('static', filename='img/tropicalis-marker.png') }}",
+    iconSize: [24, 24],
+    iconAnchor: [12, 16],
+    popupAnchor: [0, -15],
+});
+
+
+const icon_hover = L.icon({
+    iconUrl: "{{ url_for('static', filename='img/hover-icon.png') }}",
+    iconSize: [24, 24],
+    iconAnchor: [10, 16],
+    popupAnchor: [0, -15],
+});
+{% endautoescape %}
+
+  const MB_ATTR = 'Tiles &copy; Esri &mdash; Source: Esri, DeLorme, NAVTEQ, USGS, Intermap, iPC, NRCAN, METI, TomTom, 2012';
+  const MB_URL = 'https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}';
+  const OSM_URL = 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+  const OSM_ATTRIB = '&copy; <a href="https://openstreetmap.org/copyright">OpenStreetMap</a> contributors';
+
+  const southWest = L.latLng(-90, -180);
+  const northEast = L.latLng(90, 180);
+  const bounds = L.latLngBounds(southWest, northEast);
+
+  const map = L.map('map', {minZoom:2, maxBounds: bounds}).setView([20, -100], 2);
+  L.tileLayer(MB_URL, {attribution: MB_ATTR, id: 'mapbox.streets', continuousWorld: false, worldCopyJump: true}).addTo(map);
+  let strain_info = [];
+  const strain_names = [];
+  const elegans = [];
+  const briggsae = [];
+  const tropicalis = [];
+
+
+function naturalCompare(a, b) {
+    const ax = [], bx = [];
+  a = a.strain
+  b = b.strain
+
+  a.replace(/(\d+)|(\D+)/g, function(_, $1, $2) { ax.push([$1 || Infinity, $2 || ""]) });
+  b.replace(/(\d+)|(\D+)/g, function(_, $1, $2) { bx.push([$1 || Infinity, $2 || ""]) });
+  
+  while(ax.length && bx.length) {
+    const an = ax.shift();
+    const bn = bx.shift();
+    const nn = (an[0] - bn[0]) || an[1].localeCompare(bn[1]);
+      if(nn) return nn;
+  }
+
+  return ax.length - bx.length;
+}
+
+function set_marker_color (species) {
+  if (species === 'Caenorhabditis elegans') {
+    return icon_elegans;
+  }
+  if (species === 'Caenorhabditis briggsae') {
+    return icon_briggsae;
+  }
+  if (species === 'Caenorhabditis tropicalis') {
+    return icon_tropicalis;
+  }
+}
+
+const data = {{ strain_listing|tojson|safe }}
+  data.sort(naturalCompare)
+  strain_info = data;
+  data.forEach(function(d) {
+  if (d.latitude) {
+    strain_names.push(d.strain);
+    const popup_content = `
+    <div>
+      <ul class='popup'>
+        ${d.species ? `<li><strong>Species:</strong> ${d.species}</li>` : ''}
+        ${d.isotype ? `<li><strong>Isotype:</strong> ${d.isotype}</li>` : ''}
+        ${d.isotype_ref_strain ? `<li><strong>Reference Strain:</strong> ${d.isotype_ref_strain}</li>` : ''}
+        ${d.strain ? `<li><strong>Strains:</strong> ${d.strain}</li>` : ''}
+        ${d.previous_names ? `<li><strong>Alternative Names:</strong> ${d.previous_names}</li>` : ''}
+        ${d.release ? `<li><strong>Release Date:</strong> ${String(d.release).replace(/(\d{4})(\d{2})(\d{2})/, "$1-$2-$3")}</li>` : ''}
```

Comment:
instead of handling string manipulations as a one-off for the `d.release`, we should be using a dateFormat() util method/function for this. Consider creating a helper function, eg:`formatDate(d.release){...}` and move the string manipulation there. This way, in your string literal, you can replace: 

```
${d.release ? `<li><strong>Release Date:</strong> ${String(d.release).replace(/(\d{4})(\d{2})(\d{2})/, "$1-$2-$3")}</li>` : ''}
```

with:
```
${d.release ? `<li><strong>Release Date:</strong> ${ formatDate(d.release) }</li>` : ''}
```